### ISSUE TITLE
MCMT generalized gate accepts a label kwarg that seems ignored

### ISSUE THREAD

**Issue by kdk** [Qiskit/qiskit-terra#6934]
### Information

- **Qiskit Terra version**: main @ c3792f9d
- **Python version**: 3.6
- **Operating system**: osx 10.15.7

### What is the current behavior?

mcmt from library allows a label kwarg, but it doens't go anywhere?

### Steps to reproduce the problem

The generalized gate `MCMT` accepts a `label` kwarg, which it will set to be `self.label`, but as a subclass of `QuantumCircuit`, it does not look like this attribute will be read anywhere (or at least, I couldn't find anywhere where it was used.)

```
>>> from qiskit.circuit.library import *
>>> MCMT(XGate(), 3, 1, label='foo').draw()
          
q_0: ──■──
       │  
q_1: ──■──
       │  
q_2: ──■──
     ┌─┴─┐
q_3: ┤ X ├
     └───┘
>>> MCMT(XGate(), 3, 1, label='foo').name
'mcmt'
>>> MCMT(XGate(), 3, 1, label='foo').to_instruction()
<qiskit.circuit.instruction.Instruction at 0x14b031898>
>>> MCMT(XGate(), 3, 1, label='foo').to_instruction().label
>>> MCMT(XGate(), 3, 1, label='foo').to_instruction().name
'mcmt'
```

**Comment by paniash** [comment]
```py
>>> MCMT(XGate(), 3, 1, label='foo').label
foo
```

**Comment by kdk** [comment]
> ```python
> >>> MCMT(XGate(), 3, 1, label='foo').label
> foo
> ```

True, but that is equally true for any python class :). My point was more that `MCMT` accepts a `label` kwarg (like an `Instruction` subclass) but is a subclass of `QuantumCircuit`, so none of the supporting code (around drawing, converting, serializing, ...) will find this label.

**Comment by Cryoris** [comment]
I think a label doesn't make much sense for the MCMT since it's not drawn as a box but as multicontrolled gate. We could make label be the label of the controlled gate, or (probably the better, cleaner solution) just remove it.

**Comment by kevinhartman** [comment]
After some investigation, it looks like `label` is actually not ignored, though I imagine this is a bug.

When multiple target qubits are used, the name of the `QuantumCircuit` created for broadcast is set to the value of `label`. When `control()` is called on the resulting gate, that name is actually eventually used in some special-casing in `add_control.py::control`. 

```py
if gate.name == "x":
   controlled_circ.mct(q_control, q_target[bit_indices[qargs[0]]], q_ancillae)
elif gate.name == "rx":
  ...
```

I was able to get different circuits by specifying a special-cased label.

### Default (label="2-X")
```py
mcmt1 = MCMT(XGate(), num_ctrl_qubits=1, num_target_qubits=2)
print(mcmt1.qasm())
```
```qasm
OPENQASM 2.0;
include "qelib1.inc";
gate c2-X q0,q1,q2 { cx q0,q1; cx q0,q2; }
qreg q[3];
c2-X q[0],q[1],q[2];
```

### label="x"
```py
mcmt2 = MCMT(XGate(), num_ctrl_qubits=1, num_target_qubits=2, label="x")
```
```qasm
OPENQASM 2.0;
include "qelib1.inc";
qreg q[3];
cx q[0],q[1],q[2];
```

@kdk

**Comment by Cryoris** [comment]
This was addressed with #7161.
